src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,153 @@
+
+    def _atomic_nos(self):
+        """Atomic numbers of elements."""
+        return [str(Z) for Z in self.ccdata.atomnos]
+
+    def _nuclear_charges(self):
+        """Nuclear charges."""
+        return ['{:.14e}'.format(Z) for Z in self.ccdata.atomnos]
+
+    def _nuclear_coords(self):
+        """Nuclear charges."""
+        coord_template = '%.14E %.14E %.14E'        
+        nuc_coords = [coord_template % tuple(coord) for coord in self.ccdata.atomcoords[-1]]
+        return nuc_coords
+
+    def _net_charge(self):
+        """Net charge on molecule."""
+        return '%.14E' % self.ccdata.charge
+
+    def _no_electrons(self):
+        """Number of electrons in molecule."""
+        homos = self.ccdata.homos
+        charge = self.ccdata.charge
+        print (sum(homos), charge)
+        if len(homos) > 1:
+            return int(sum(homos))
+        elif charge > 0:
+            return int(homos[0]*2 - (charge % 2))
+
+    def generate_repr(self):
+        """Generate the wfx representation of the logfile data."""
+
+        sections = [
+            (self._title, "Title"),
+            (self._keywords, "Keywords"),
+            (self._no_of_prims, "Number of Primitives"),
+            (self._no_of_mos, "Number of Occupied Molecular Orbitals"),
+            (self._no_of_perturbations, "Number of Perturbations"),
+            (self._nuclear_names, "Nuclear Names"),
+            (self._atomic_nos, "Atomic Numbers"),
+            (self._nuclear_charges, "Nuclear Charges"),
+            (self._nuclear_coords, "Nuclear Cartesian Coordinates"),
+            (self._net_charge, "Net Charge"),
+            (self._no_electrons, "Number of Electrons"),
+            # (self._no_alpha_electrons, "Number of Alpha Electrons"),
+            # (self._no_beta_electrons, "Number of Beta Electrons"),
+            # (self._spin_mult, "Electronic Spin Multiplicity"),
+            # (self._model, "Model"),
+            # (self._prim_centers, "Primitive Centers"),
+            # (self._prim_types, "Primitive Types"),
+            # (self._prim_exps, "Primitive Exponents"),
+            # (self._mo_occup_nos, "Molecular Orbital Occupation Numbers"),
+            # (self._mo_energies, "Molecular Orbital Energies"),
+            # (self._mo_spin_types, "Molecular Orbital Spin Types"),
+            # (self._mo_prim_coeffs, "Molecular Orbital Primitive Coefficients"),
+            # (self._energy, "Energy"),
+            # (self._nuc_energy_gradients, "Nuclear Cartesian Energy Gradients"),
+            # (self._nuc_virial, "Nuclear Virial of Energy-Gradient-Based Forces on Nuclei, W"),
+            # (self._virial_ratio, "Full Virial Ratio"),
+        ]
+
+        wfx_lines = []
+
+        for section_module, section_name in sections:
+            try:
+                section_data = section_module()
+            except:
+                section_data = None
+            wfx_lines.extend(self._section(section_name, section_data))
```

Comment:
Thats a good idea. This particular portion is a hurried job after my internal refactoring.